cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -350,6 +351,22 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		serviceaccountcontroller.DefaultServiceAccountsControllerOptions(),
 	).Run()
 
+	if s.HealthCheckControllerSyncPeriod > 0 {
+		config := client.AddUserAgent(kubeconfig, "healthchecker-controller")
+
+		checkerTransport, err := client.TransportFor(config)
+		if err != nil {
+			panic(err)
+		}
+		httpClient := &http.Client{Transport: checkerTransport}
+		healthChecker := healthzcheckcontroller.NewHTTPSHealthzChecker(httpClient, "healthz")
+
+		healthCheckClient := clientset.NewForConfigOrDie(config)
+		checkController := healthzcheckcontroller.NewAPIServerHealthCheckController(
+			healthCheckClient.Core().Endpoints("default"), healthChecker, "kubernetes", recorder)
```

Comment:
This assumes that the controller can reach the apiservers on this address - you should note that in the godoc.